valueonly/context.go

```diff
@@ -0,0 +1,13 @@
+package valueonly
+
+import (
+	"context"
+	"time"
+)
+
+// Context that wraps another and suppresses its deadline or cancellation.
```

Comment:
My two cents is that this is a risky pattern that we should not encourage. How do we prevent a zombie go routine? 

If the goal is to copy trace headers from one context to another, I would argue a safer implementation is to update the o11y interface to include https://github.com/honeycombio/beeline-go/blob/main/trace/context.go#L65.

Reply:
Seems reasonable to me!

Reply:
Should we go this way instead?

Reply:
It's certainly a bigger change, but definitely something useful for `o11y`. Again, it depends on how much time you have for this. You'd need to make sure you copied the `o11y` provider and span, at least. I don't *think* there's more, but you'd have to write some tests to be sure.

Reply:
Personally I think this risk is about the same with both methods. As whichever way you do it, you'll be creating a new context to put stuff in, and will need to remember to either wrap it in a context with a timeout, or make sure whatever you're doing inside internally has a timeout.

Reply:
If you carry on with the value only context approach, you should add a constructor for it taking the parent context.

Reply:
I think we should do like Shaun says, but the https://github.com/circleci/execution-gateway/pull/38 should not be made dependent on it.

I think we could add out own deadline at the beginning of the goroutine there https://github.com/circleci/execution-gateway/pull/38/files#diff-e190276d77368b3d2fada2abc3aa6ece46a82850c31e69a0365855241abd29bbR41

Reply:
> Personally I think this risk is about the same with both methods.

> If you carry on with the value only context approach, you should add a constructor for it taking the parent context.

I saw these comments now.

It also seems good. We could make the value-only context constructor apply a mandatory timeout.

I understand we could use this pattern https://blog.j7mbo.com/bypassing-golangs-lack-of-constructors/